modules/Sources/Features/SendFlow/SendFlowStore.swift

```diff
@@ -67,47 +67,46 @@ public struct SendFlowReducer: Reducer {
         }
 
         public var isInvalidAddressFormat: Bool {
-            !transactionAddressInputState.isValidAddress
-            && !transactionAddressInputState.textFieldState.text.data.isEmpty
+            !transactionAddressInputState.textFieldState.text.data.isEmpty
+            && !transactionAddressInputState.isValidAddress
         }
 
         public var isInvalidAmountFormat: Bool {
-            !transactionAmountInputState.textFieldState.valid
-            && !transactionAmountInputState.textFieldState.text.data.isEmpty
+            !transactionAmountInputState.textFieldState.text.data.isEmpty
+            && !transactionAmountInputState.isValidInput
         }
         
         public var isValidForm: Bool {
-            @Dependency(\.zcashSDKEnvironment) var zcashSDKEnvironment
-
-            return transactionAmountInputState.amount.data > zcashSDKEnvironment.network.constants.defaultFee().amount
-            && transactionAddressInputState.isValidAddress
+            !isInvalidAddressFormat
             && !isInsufficientFunds
             && memoState.isValid
+            && !isInvalidAmountFormat
         }
-        
+
         public var isInsufficientFunds: Bool {
-            transactionAmountInputState.amount.data > transactionAmountInputState.maxValue.data
+            @Dependency(\.zcashSDKEnvironment) var zcashSDKEnvironment
+
+            return transactionAmountInputState.amount.data > spendableBalance.amount - zcashSDKEnvironment.network.constants.defaultFee().amount
         }
         
         public var isMemoInputEnabled: Bool {
-            transactionAddressInputState.textFieldState.text.data.isEmpty ||
             !transactionAddressInputState.isValidTransparentAddress
```

Comment:
What happens if the user enters something into the memo field and then enters a valid transparent address? Is the memo entry grayed out? Does it give an error if the user then clicks send?

Reply:
Oh I see, if I'm reading correctly then it will clear the memo field in that case.

Reply:
All I can do is to confirm all you've written. Redundancy + clear of the memo in case the valid tAddress is used. THe clearance has been defined by product team and Josh.